src/python_inspector/package_data.py

```diff
@@ -67,16 +67,17 @@ def get_pypi_data_from_purl(
     # if prefer_source is True then only source distribution is used
     # in case of no source distribution available then wheel is used
     if not valid_distribution_urls or not prefer_source:
-        valid_distribution_urls.extend(
-            list(
-                get_wheel_download_urls(
-                    purl=purl,
-                    repos=repos,
-                    environment=environment,
-                    python_version=python_version,
-                )
+        wheel_urls = list(
+            get_wheel_download_urls(
+                purl=purl,
+                repos=repos,
+                environment=environment,
+                python_version=python_version,
             )
         )
+        wheel_urls.sort(reverse=True)
+        if len(wheel_urls) > 0:
+            valid_distribution_urls.append(wheel_urls[0])
```

Comment:
Move this logic in a function, such as choose_single_wheel, to make the code more explicit.
Make sure to add a doctring too.

Reply:
Done!